### PR TITLE
Toggle displays based on weekly exercise log completion.

### PR DESCRIPTION
**Pull request by eschlange**
* Added TimeCop gem for testing.
* Updated home and exercise log controllers.
* Added specs for new controller logic.
* Added scopes to exercise log and response objects.
* Updated JS files with new logic to hide and show exercise log data.

[#111698532]

### CODE REVIEW

File: app/assets/javascripts/assignments.js [hunk 1]
```diff
@@ -73,6 +73,13 @@ function todoList(current_user_id) {
 							toggleTask(current_user_id, task.id)
 						});
     		});
+
+        if(!is_exercise_log_complete) {
+          var log_link = "<li class='layer'><ul><li><a href='user/" + current_user_id + "/exercise_logs'>" +
```
Comment (ericcf) [inline app/assets/javascripts/assignments.js]: aaaah, my eyes are bleeding :eyes:

Comment (eschlange) [inline app/assets/javascripts/assignments.js]: The blood loss is mutual.

File: app/controllers/exercise_logs_controller.rb [hunk 2]
```diff
@@ -3,18 +3,25 @@ class ExerciseLogsController < ApplicationController
```
Comment (ericcf) [inline app/controllers/exercise_logs_controller.rb]: I know this seems petty, but would you mind fixing the inconsistent formatting? This is nearly impossible to read in GitHub. Thanks.

Comment (eschlange) [inline app/controllers/exercise_logs_controller.rb]: I'm not sure why it is formatted this way in the Github view. There aren't any errant spacing issues. I'll remove the line breaks to see if it makes a difference.

File: app/models/response.rb [hunk 3]
```diff
@@ -20,6 +20,37 @@ def weeks_into_assigment
     end
   end
 
+  scope :with_user, -> (user) { where(user: user) }
+  scope :created_after, -> (time) { where("created_at >= ?", time) }
+  scope :created_before, -> (time) { where("created_at <= ?", time) }
+  scope :includes_flash_cards, -> { where('flash_card_id IS NOT NULL') }
```
Comment (ericcf) [inline app/models/response.rb]: You could use the ActiveRecord dsl here: https://robots.thoughtbot.com/activerecords-wherenot

Comment (eschlange) [inline app/models/response.rb]: I like it, much cleaner syntax. Making the update now.

File: app/controllers/exercise_logs_controller.rb (hunk 2 above)
Comment (eschlange) [inline app/controllers/exercise_logs_controller.rb]: Removing the line breaks makes it much more readable. The problem may have stemmed from Rubymine hiding tabs.

**Comment by ericcf** [comment]
Thanks.

**Pull request merged**